internal/dataplane/failures/failures.go

```diff
@@ -41,9 +41,13 @@ func NewResourceFailure(reason string, causingObjects ...client.Object) (Resourc
 		if obj.GetName() == "" {
 			return ResourceFailure{}, fmt.Errorf("one of causing objects (%s) has no name", gvk.String())
 		}
-		if obj.GetNamespace() == "" {
-			return ResourceFailure{}, fmt.Errorf("one of causing objects (%s) has no namespace", gvk.String())
-		}
+		// REVIEW: should we remove this to allow generating resource failures for cluster scope objects like KongClusterPlugins?
+		// https://github.com/Kong/kubernetes-ingress-controller/issues/5387
+		/*
+			if obj.GetNamespace() == "" {
+				return ResourceFailure{}, fmt.Errorf("one of causing objects (%s) has no namespace", gvk.String())
+			}
+		*/
```

Comment:
Does Kubernetes allow global Events? I didn't think it did; I don't know where we'd put them. I guess we could stick them in the controller's namespace, for lack of a better location, if there is indeed no ClusterEvent or whatever (there's no resource named that specifically, bu maybe there's something else?).

https://github.com/Kong/kubernetes-ingress-controller/issues/5387#issuecomment-1879497941 for some other thoughts.

For now I think we'd uncomment the original code here in place until #5387 has an implementation

Reply:
removed the change here.